File: LeetCode/ConstructAlgorithm/Golang/Leetcode1656-design-an-ordered-stream.go
package sbgo

type OrderedStream struct {
	mp  map[int]string
	ptr int
}

func Constructor(n int) OrderedStream {
	mp := map[int]string{}
	return OrderedStream{mp, 1}
}

func (this *OrderedStream) Insert(idKey int, value string) []string {
	this.mp[idKey] = value
	ans := []string{}
	if _, ok := this.mp[this.ptr]; !ok {
		return ans
	}
	for value, ok := this.mp[this.ptr]; ok; value, ok = this.mp[this.ptr] {
		if !ok {
			break
		}
		ans = append(ans, value)
		this.ptr++
	}
	return ans
}

File: LeetCode/ConstructAlgorithm/PY/Leetcode1656-design-an-ordered-stream.py
class OrderedStream:

    def __init__(self, n: int):
        self.mp = defaultdict(str)
        self.ptr = 1


    def insert(self, idKey: int, value: str) -> List[str]:
        self.mp[idKey] = value
        ans = []
        if self.ptr not in self.mp: return ans
        while self.ptr in self.mp:
            ans.append(self.mp[self.ptr])
            self.ptr += 1
        return ans
